scipy/stats/_stats_py.py

```diff
@@ -670,7 +668,7 @@ def tmean(a, limits=None, inclusive=(True, True), axis=None):
     # explicit dtype specification required due to data-apis/array-api-compat#152
     sum = xp.sum(a, axis=axis, dtype=a.dtype)
     n = xp.sum(xp.asarray(~mask, dtype=a.dtype), axis=axis, dtype=a.dtype)
-    mean = _lazywhere(n != 0, (sum, n), xp.divide, xp.nan)
+    mean = xpx.apply_where(n != 0, operator.truediv, (sum, n), fill_value=xp.nan)
```

Comment:
Explanation here: https://github.com/data-apis/array-api-extra/pull/141/files#r1961964801

Reply:
Haven't studied the reason yet, but change looks fine.